src/org/zaproxy/zap/extension/brk/BreakAPI.java

```diff
@@ -91,6 +116,60 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
 		} else if (ACTION_BREAK_ON_ID.equals(name)) {
 			extension.setBreakOnId(params.getString(PARAM_KEY), params.getString(PARAM_STATE).toLowerCase().equals("on"));
 
+		} else if (ACTION_CONTINUE.equals(name)) {
+			extension.getBreakpointManagementInterface().cont();
+
+		} else if (ACTION_STEP.equals(name)) {
+			extension.getBreakpointManagementInterface().step();
+
+		} else if (ACTION_DROP.equals(name)) {
+			extension.getBreakpointManagementInterface().drop();
+
+		} else if (ACTION_SET_HTTP_MESSAGE.equals(name)) {
+			if (extension.getBreakpointManagementInterface().getMessage() == null) {
+				// We've not got an intercepted message
+				throw new ApiException(ApiException.Type.DOES_NOT_EXIST);
+			}
+
+			String header = params.getString(PARAM_HTTP_HEADER);
+			String body = this.getParam(params, PARAM_HTTP_BODY, "");
+
+			if (header.indexOf(HttpHeader.CRLF) < 0) {
+				if (header.indexOf("\\n") >= 0) {
+					// Makes it easier to use via API UI
+					header = header.replace("\\r", "\r").replace("\\n", "\n");
+				}
+			}
+
+			if (extension.getBreakpointManagementInterface().isRequest()) {
+				HttpMessage msg = new HttpMessage();
+	
+				try {
+					msg.setRequestHeader(header);
+					msg.setRequestBody(body);
+					extension.getBreakpointManagementInterface().setMessage(msg, true);
+					extension.getBreakpointManagementInterface().saveMessage(true);
+	
+				} catch (HttpMalformedHeaderException e) {
+					throw new ApiException(ApiException.Type.ILLEGAL_PARAMETER, e.getMessage());
+				}
+			} else {
+				Message msg = extension.getBreakpointManagementInterface().getMessage();
+				if (msg instanceof HttpMessage) {
+					HttpMessage httpMsg = (HttpMessage)msg;
+		
+					try {
+						httpMsg.setResponseHeader(header);
+						httpMsg.setResponseBody(body);
+						extension.getBreakpointManagementInterface().setMessage(msg, false);
+						extension.getBreakpointManagementInterface().saveMessage(false);
```

Comment:
This call is not needed as the message is already being saved (in `BreakpointMessageHandler2.waitUntilContinue`) before being sent?